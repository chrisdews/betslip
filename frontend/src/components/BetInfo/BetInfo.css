
.betInfoContainer {
    display: flex;
    justify-content: space-between;
    align-items: center;   
    border: 1px solid grey;
    padding: 2em;
    margin: 1em;
}

.betInfoContainer > .betDetailContainer {
    flex-direction: column;
}