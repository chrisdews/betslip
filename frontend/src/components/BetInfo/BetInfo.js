import React from "react";
import PropTypes from "prop-types";
import StakeInput from "../../components/StakeInput";
import styled from "styled-components";

function BetInfo({ betName, bestOdds, betBuilder }) {
  return (
    <BetInfoContainer>
      <BetDetailContainer>
        <div>{betName} </div>
        <div>{bestOdds}</div>
      </BetDetailContainer>
      <StakeInput
        bestOdds={bestOdds}
        betBuilder={betBuilder}
        betName={betName}
      />
    </BetInfoContainer>
  );
}

BetInfo.propTypes = {
  betName: PropTypes.string,
  bestOdds: PropTypes.number,
  betBuilder: PropTypes.func,
};

export default BetInfo;

const BetInfoContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  border: 2px solid white;
  border-radius: 1em;
  padding: 2em;
  margin: 1em;
  background-color: rgb(0, 24, 46);
`;

const BetDetailContainer = styled.div`
  flex-direction: column;
`;
