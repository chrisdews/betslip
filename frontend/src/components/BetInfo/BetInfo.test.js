import React from "react";
import BetInfo from "./BetInfo";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("BetInfo", () => {
  it("renders correctly - snapshot", () => {
    const { container } = render(<BetInfo betName={'dewsy'} bestOdds={3.14}/>);
    expect(container).toMatchSnapshot();
  });
});