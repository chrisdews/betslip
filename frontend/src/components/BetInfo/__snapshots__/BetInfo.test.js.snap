// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetInfo renders correctly - snapshot 1`] = `
<div>
  <div
    class="sc-gtsrHT hisShj"
  >
    <div
      class="sc-dlnjwi cEfaSu"
    >
      <div>
        dewsy
         
      </div>
      <div>
        3.14
      </div>
    </div>
    <input
      aria-label="stake-input"
      class="sc-bdnxRM liqCke"
      type="text"
      value="enter stake"
    />
  </div>
</div>
`;
